src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSnapshotManagerDispatcher.DefaultScheduler.cs

```diff
@@ -64,18 +59,20 @@ private void ThreadStart()
                 {
                     try
                     {
-                        if (_taskQueue.TryTake(out var task, Timeout.Infinite, disposeToken))
-                        {
-                            TryExecuteTask(task);
-                        }
+                        // ConfigureAwait(true) is used below because this method runs on the
+                        // ThreadPool and we want to continue running on the ThreadPool.
```

Comment:
This comment is surprising, or my understanding is lacking. Is it not correct that `ConfigureAwait(true)` will mean continue running on the same thread, whereas `ConfigureAwait(false)` would still have us continue on the ThreadPool?

Reply:
Not exactly. `ConfigureAwait(false)` means that we don't want to capture the sync context to run the continuation. In practice, that means that it'll run on the thread pool. Since we're already on the thread pool here, it really doesn't matter - capturing the sync context just puts us back on the thread pool. It'd likely be better to just not use `ConfigureAwait(...)` at all and enjoy the code brevity. :smile:

FWIW, in Visual Studio, it is recommended _not_ to use `ConfigureAwait(false)` except in app-independent library code, such as Roslyn. https://github.com/microsoft/vs-threading/blob/main/doc/cookbook_vs.md#should-i-await-a-task-with-configureawaitfalse

Reply:
I'm going to tweaking this a bit yet, so I'll likely remove this comment.